rust/pattern/constraint/mod.rs

```diff
@@ -28,3 +28,9 @@ pub use type_constraint::*;
 
 mod thing_constraint;
 pub use thing_constraint::*;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum IsExplicit {
+    Yes,
+    No,
+}
```

Comment:
Shared between `IsaConstraint` and `SubConstraint`--it serves the same purpose for both--, so it's at the root of the module.